File: supabase_client.py
"""
Supabase 客户端连接模块
"""
import os
from supabase import create_client, Client
from config import config

def create_supabase_client() -> Client:
    """
    创建 Supabase 客户端

    Returns:
        Client: Supabase 客户端实例

    Raises:
        ValueError: 当配置不完整时抛出
    """
    if not config.validate():
        raise ValueError("Supabase 配置不完整，请检查环境变量")

    print(f"连接到 Supabase: {config.url}")
    return create_client(config.url, config.anon_key)

def create_supabase_admin_client() -> Client:
    """
    创建 Supabase 管理员客户端（使用 service role key）

    Returns:
        Client: Supabase 管理员客户端实例

    Raises:
        ValueError: 当配置不完整时抛出
    """
    if not config.service_role_key:
        raise ValueError("Service role key 未配置")

    print(f"连接到 Supabase (管理员模式): {config.url}")
    return create_client(config.url, config.service_role_key)

# 全局客户端实例
supabase: Client = create_supabase_client()

if __name__ == "__main__":
    # 测试连接
    try:
        # 测试基本连接
        response = supabase.table('_supabase_tables').select('*').limit(1).execute()
        print("✅ Supabase 连接成功")
        print(f"响应: {response}")
    except Exception as e:
        print(f"❌ Supabase 连接失败: {e}")
        print("请检查您的配置和网络连接")

File: supabase-client.js
/**
 * Supabase 客户端连接模块 (Node.js版本)
 */
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

// 从环境变量获取配置
const supabaseUrl = process.env.SUPABASE_URL;
const supabaseAnonKey = process.env.SUPABASE_ANON_KEY;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

// 验证配置
if (!supabaseUrl || !supabaseAnonKey) {
  throw new Error('缺少必要的环境变量：SUPABASE_URL 或 SUPABASE_ANON_KEY');
}

// 创建客户端实例
const supabase = createClient(supabaseUrl, supabaseAnonKey);
const supabaseAdmin = supabaseServiceKey ? createClient(supabaseUrl, supabaseServiceKey) : null;

console.log(`连接到 Supabase: ${supabaseUrl}`);

/**
 * 获取基础客户端（使用anon key）
 */
function getSupabaseClient() {
  return supabase;
}

/**
 * 获取管理员客户端（使用service role key）
 */
function getSupabaseAdminClient() {
  if (!supabaseAdmin) {
    throw new Error('Service role key 未配置，无法创建管理员客户端');
  }
  return supabaseAdmin;
}

/**
 * 测试数据库连接
 */
async function testConnection(client = supabase) {
  try {
    console.log('测试 Supabase 连接...');

    // 尝试一个简单的查询来测试连接
    // 注意：这个查询可能因为权限或表不存在而失败，但可以验证连接本身
    const { data, error } = await client
      .from('_supabase_tables')
      .select('*')
      .limit(1);

    if (error) {
      console.log(`连接测试完成，但查询失败: ${error.message}`);
      console.log('这可能是正常的，因为测试表可能不存在或权限不足');
      return true; // 连接成功，只是查询失败
    }

    console.log('✅ Supabase 连接测试成功');
    return true;
  } catch (err) {
    console.error('❌ Supabase 连接测试失败:', err.message);
    return false;
  }
}

module.exports = {
  supabase,
  supabaseAdmin,
  getSupabaseClient,
  getSupabaseAdminClient,
  testConnection
};

// 如果直接运行此文件，则执行连接测试
if (require.main === module) {
  testConnection().then(success => {
    if (success) {
      console.log('🎉 Supabase 客户端配置完成');
      process.exit(0);
    } else {
      console.error('❌ Supabase 客户端配置失败');
      process.exit(1);
    }
  });
}
